module/input_handler.py

```diff
@@ -64,70 +51,154 @@ def convert_type(convertable, convert_type):
+        )
+        print(
+            "\tGlobal commands: {}".format(
+                ", ".join(
+                    map(str.lower, self.cli_controller.global_commands.keys())
+                )
+            )
+        )
+
+        if self.cli_controller.target:
+            print(
+                "\n\tCurrently selected {}: {}".format(
+                    self.cli_controller.target[1].name.lower(), self.cli_controller.target[0].lower()
+                )
+            )
+            print(
+                "\tTarget specific commands: {}".format(
+                    ", ".join(map(str.lower, self.cli_controller.target[1].keys()))
+                )
+            )
+        else:
+            print("\n\tNo target selected.")
+
+    def _handle_category_command(self, command: str, params: list) -> None:
+        """
+        Handles all non-global commands. Returns false if failed or
+        if a function has been executed(in this case no other commands can be executed after).
+        Returns true if another command can be executed after this one
+        """
+        category = self.cli_controller.target[1]
+
+        if command not in category:
+            self._print_command_not_found(command)
+            return
+
+        required_params = category[command]
+
+        if not self._check_amount_parameters(params, len(required_params)):
+            return
+
+        # Validate if the user paramters are of the correct type.
+        # Print invalid type if type is invalid, this code evaluates all parameters.
+        correct_params = True
+        for i, param in enumerate(required_params):
+            try:
+                params[i] = self._convert_type(params[i], required_params[param])
+            except ValueError:
+                print(
+                    "\tInvalid type for parameter {} '{}', expected '{}'".format(
+                        i, param, required_params[param]
+                    )
+                )
+                correct_params = False
+
+        if not correct_params:
+            return
+
+        print("\tSending command:", command.lower(), params, " to: ", self.cli_controller.target[0].lower())
+        # cant be executed yet as python bus string frames are not working like intended yet
+        #category[command].send(
+        #    self.cli_controller.comm, params, self.cli_controller.target[0]
+        #)
+
+    def _handle_command(self, command: str, params: list):
+        if command in self.cli_controller.global_commands:
+            self.cli_controller.global_commands[command].execute(params)
+        elif self.cli_controller.target:
+            self._handle_category_command(command, params)
+        else:
+            self._print_command_not_found(command)
+            print(
+                """\tMaybe the command you were trying to execute is a target specific command,
+\tuse 'select' to select a target."""
+            )
+
+    def handle_input(self, input_words: list) -> None:
         """
```

Comment:
Please document the parameter.

Reply:
Please us @param input_words when commenting the parameter.

Reply:
params is a list containing strings of parameters||

This is still very vague for me, but this is your own choice. So i approve the changes made.